src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -86,24 +104,85 @@ def add_helm_repo(kube_config, kube_context):
         raise CLIError("Unable to add repository {} to helm: ".format(repo_url) + error_helm_repo.decode("ascii"))
 
 
-def get_helm_registry(profile, location):
-    cred, _, _ = profile.get_login_credentials(
-        resource='https://management.core.windows.net/')
-    token = cred._token_retriever()[2].get('accessToken')  # pylint: disable=protected-access
-
+def get_helm_registry(cmd, location):
     get_chart_location_url = "https://{}.dp.kubernetesconfiguration.azure.com/{}/GetLatestHelmPackagePath?api-version=2019-11-01-preview".format(location, 'azure-arc-k8sagents')
-    query_parameters = {}
-    query_parameters['releaseTrain'] = os.getenv('RELEASETRAIN') if os.getenv('RELEASETRAIN') else 'stable'
-    header_parameters = {}
-    header_parameters['Authorization'] = "Bearer {}".format(str(token))
+    release_train = os.getenv('RELEASETRAIN') if os.getenv('RELEASETRAIN') else 'stable'
+    uri_parameters = ["releaseTrain={}".format(release_train)]
     try:
-        response = requests.post(get_chart_location_url, params=query_parameters, headers=header_parameters)
+        r = send_raw_request(cmd.cli_ctx, 'post', get_chart_location_url, uri_parameters=uri_parameters, resource='https://management.core.windows.net/')
```

Comment:
pls not hardcode the url, but get from azure cli core.

Reply:
you can use `cmd.cli_ctx.cloud.endpoints.management`

Reply:
I have updated the resource url to fetch from azure cli core. If you meant 'get_chart_location_url', we cannot get it from azure cli core as it is dataplane url and not present in arm

Reply:
Yes, I have used cmd.cli_ctx.cloud.endpoints.management